go/obscuronode/enclave/enclave.go

```diff
@@ -419,18 +450,28 @@ type processingEnvironment struct {
 	processedTxsMap map[common.Hash]nodecommon.L2Tx
 }
 
-func NewEnclave(id common.Address, mining bool, collector StatsCollector) nodecommon.Enclave {
+// NewEnclave creates a new enclave.
+// `genesisJSON` is the configuration for the corresponding L1's genesis block. This is used to validate the blocks
+// received from the L1 node. If nil, incoming blocks are not validated.
+func NewEnclave(id common.Address, mining bool, genesisJSON []byte, collector StatsCollector) nodecommon.Enclave {
 	storage := NewStorage()
+
+	var l1Blockchain *core.BlockChain
+	if genesisJSON != nil {
```

Comment:
I don't like this implicit mechanism of disabling validation.
I'd suggest an explicit mechanism with a configuration flag, and then if "validation==true" then panic if there's no genesisJson